discovery-infra/test_infra/cluster.py

```diff
@@ -0,0 +1,237 @@
+import logging
+
+from tests.conftest import env_variables
+from test_infra import consts, utils
+
+
+class Cluster:
+    
+    def __init__(self, api_client, cluster_name):
+        res = self._create(api_client, cluster_name)
+
+        self.id = res.id
+        self.api_client = api_client
+    
+    def _create(self, api_client, cluster_name):
```

Comment:
if we have api client in init , why we need to provide it to every function?

Reply:
I need to be able override the client for auth tests